### ISSUE TITLE
words extracted from pdf getting split

### ISSUE THREAD

**Issue by nischithac** [pymupdf/PyMuPDF#1159]
_**Please provide all mandatory information!**_

## Describe the bug (mandatory)
Few words are getting split into half . Example beneﬁciaries  word is split into 'beneﬁ', 'ciaries' . But there is no space between this words in pdf 

## To Reproduce (mandatory)
doc = fitz.open(input_file)
page = doc[0]
page.getText()
Output: The discounts are not available to beneﬁ ciaries of \nMedicare, Medicaid or other federal or state healthcare programs or residents of Massachusetts, Puerto Rico and other US territories.

## Expected behavior (optional)
words shouldn't get split. 
Expected Output: The discounts are not available to beneﬁciaries of \nMedicare, Medicaid or other federal or state healthcare programs or residents of Massachusetts, Puerto Rico and other US territories.

## Your configuration (mandatory)
 - Operating system, potentially version and bitness: Ubuntu
 - Python version, bitness: Python 3.6
 - PyMuPDF version, installation method (**wheel** or **generated** from source). Pymupdf 1.18.9

For example, the output of `print(sys.version, "\n", sys.platform, "\n", fitz.__doc__)` would be sufficient (for the first two bullets).

**Comment by JorjMcKie** [comment]
Please:
* upgrade to the current version
* provide minimal example file to reproduce
* read the documentation concerning text extraction flag `TEXT_INHIBIT_SPACES` and friends.
* try [this](https://github.com/pymupdf/PyMuPDF-Utilities/blob/master/text-extraction/textlayout.py) layout-preserving script.

**Comment by nischithac** [comment]
I tried all the solutions which you mentioned  but its not working. This is a sample pdf file where the words 'first', 'notifications' are getting split. 
[Split_word_issue.pdf](https://github.com/pymupdf/PyMuPDF/files/6862551/Split_word_issue.pdf)

**Comment by JorjMcKie** [comment]
This is a badly constructed PDF:
Invisibly for the PDF viewer, space characters are specified which (partly) overlap their preceeding character. E.g. "Notifi" ends at some x-coordinate, say 25.5, then a space character follows which sparts at 25.0 ending at 26.0, followed by "cations" ...
I have modified the script `textlayout.py` such that it detects the situation and ignores those spaces. I have attached it here.
Play with it until you see acceptable results.
It definitely is not a bug of PyMuPDF, but the PDF maker screwed up the file.
[textlayout.zip](https://github.com/pymupdf/PyMuPDF/files/6863617/textlayout.zip)
